tiledb/sm/query/query.h

```diff
@@ -918,6 +918,13 @@ class Query {
    */
   RestClient* rest_client() const;
 
+  /**
+   * Returns the ContextResources associated to this query.
+   */
```

Comment:
We should remove this. It was added as a way to get the serialization memory tracker but we should be able to get it from another source.

Reply:
As said above, query is for me the best candidate to get the allocator.